src/elements/toniq-slider/toniq-slider.element.ts

```diff
@@ -0,0 +1,407 @@
+         */
+        max: 100,
+        /** Set to true to enable double range slider. */
+        double: false,
+        /** Use to add suffix to the value. */
+        suffix: '',
+    },
+    events: {
+        valueChange: defineElementEvent<number | ToniqSliderDoubleRangeValue>(),
+    },
+    styles: css`
+        .range {
+            display: flex;
+            height: 8px;
+            width: 100%;
+            border-radius: 4px;
+            margin: 0;
+            margin-bottom: 48px;
+            ${applyBackgroundAndForeground(toniqColors.accentSecondary)};
+        }
+
+        .progress {
+            position: absolute;
+            height: 8px;
+            z-index: 1;
+            pointer-events: none;
+            border-radius: 4px;
+            ${applyBackgroundAndForeground(toniqColors.accentPrimary)};
+        }
+
+        .label {
+            z-index: 1;
+            position: absolute;
+            top: 16px;
+            ${toniqFontStyles.boldParagraphFont};
+            ${noUserSelect};
+        }
+
+        .slider {
+            -webkit-appearance: none;
+            height: 8px;
+            width: 100%;
+            background-color: transparent;
+            margin: 0;
+        }
+
+        #lowerSlider,
+        #upperSlider {
+            position: absolute;
+            left: 0;
+            pointer-events: none;
+        }
+
+        .slider-wrapper {
+            position: relative;
+            width: 100%;
+        }
+
+        .slider::-webkit-slider-thumb {
+            -webkit-appearance: none;
+            position: relative;
+            height: 16px;
+            width: 16px;
+            z-index: 10;
+            cursor: pointer;
+            pointer-events: all;
+            border-radius: 10px;
+            transition: ${interactionDuration};
+            ${applyBackgroundAndForeground(toniqColors.accentPrimary)};
+        }
+
+        .slider::-webkit-slider-thumb:hover {
+            transform: scale(1.2);
+        }
+    `,
+    initCallback: ({props, setProps}) => {
+        setProps({value: props.value < props.min ? props.min : props.value});
+    },
+    renderCallback: ({props, host, dispatch, events}) => {
+        function onSliderResize(
+            entry: Readonly<Pick<ResizeObserverEntry, 'target' | 'contentRect'>>,
+        ) {
+            if (props.double) {
+                fillDoubleRangeSlider();
+            } else {
+                fillRangeSlider(parseInt((entry.target as HTMLInputElement).value));
+            }
+        }
+
+        function fillRangeSlider(value: number) {
+            const slider = host.shadowRoot?.querySelector('.slider') as HTMLInputElement;
+            const progress = host.shadowRoot?.querySelector('.progress') as HTMLElement;
+            const label = host.shadowRoot?.querySelector('.label') as HTMLElement;
+            const sliderWidth = slider.clientWidth;
+
+            const sliderOffset = (8 * value) / props.max;
```

Comment:
What is `8` here? It should be saved into a variable with a name that indicates what it's for.